src/main/java/io/ipfs/api/IPFS.java

```diff
@@ -490,6 +492,28 @@ public Object log() throws IOException {
         }
     }
 
+    private boolean versionIsValid(String version) {
+        String[] parts = version.split("\\.");
```

Comment:
Hi @VictorBjelkholm! All of our application logic is written in Java. Originally we had a parallel implementation in JS but it was so much slower to write correctly and error prone. Writing the original in Java and cross compiling lets us move much faster and with much more safety and guarantees. We do have a few interfaces where java code can call out to JS (or Java if it's not running in a browser), but we try to keep those to a minimum. 

We are also extremely sensitive to importing random libraries, for security reasons. I have never built js-ipfs-api, but I would guess it brings in 100s of dependencies through npm. All of these would have to be audited in detail in our security audit.